readers/mutation_reader.cc

```diff
@@ -385,11 +385,6 @@ flat_mutation_reader_v2::~flat_mutation_reader_v2() {
     }
 }
 
-void flat_mutation_reader_v2::impl::forward_buffer_to(const position_in_partition& pos) {
-    clear_buffer();
-    _buffer_size = compute_buffer_size(*_schema, _buffer);
-}
```

Comment:
This smells like a bug to me. This method was supposed to clear buffer *up to* the specified position.
Did we drop this on the floor with the v2 migration? @tgrabiec?

Reply:
It was needed with flat_mutation_reader which had range_tombstone fragments. They could be relevant to the next position_range on fast_forward_to() because they could hold information which extends past the current range. With flat_mutation_reader_v2 we no longer have such fragments, so mutation_fragment::relevant_for_range_assuming_after() was dropped and this is effectively clearing all. We just forgot to remove it. So I think this change is fine.